lib/superform.rb

```diff
@@ -105,7 +105,7 @@ def initialize(key, parent:, object: nil, field_class: Field)
     # end
     # ```
     def namespace(key, &block)
-      create_child(key, self.class, object: object_for(key: key), &block)
+      create_child(key, self.class, field_class: @field_class, object: object_for(key: key), &block)
```

Comment:
I don't think this has to be tackled in this commit, but the `create_child` method is probably too complex, which is why I'm having a hard time reasoning about it in this PR (and I wrote the code 🤣). What's tripping me up is seeing `self.class` passed into this method and `field_class`. It's hiding the fact that a Namespace class is being created and registered as a child, so there's really probably three methods that could be extracted to improve readability here.

The methods hiding here are probably `create_child_namespace` and `create_child_field` method, both of which call the `create_child` method to register itself in the hash.

Reply:
Also I think `attr_reader :field_class` needs to be added so that it can be overridden via inheritance. You'd then pass `field_class: field_class` into the keyword args.